did/plugins/phabricator.py

```diff
@@ -0,0 +1,274 @@
+                raise ReportError(
+                    "Phabricator failed to parse JSON response") from error
+
+            res = decoded["result"]
+            result.extend(Differential(diff) for diff in res["data"])
+            # Define offset of next differentials to fetch
+            data_dict['after'] = res["cursor"]["after"]
+            if data_dict['after'] is None:
+                break
+
+        log.debug("Result: %s fetched", listed(len(result), "item"))
+        log.data(pretty(result))
+        return result
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  Differential
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+class Differential:  # pylint: disable=too-few-public-methods
+    """
+    Phabricator Differential
+
+    Here's an example::
+
+        {
+            "id": 134852,
+            "type": "DREV",
+            "phid": "PHID-DREV-6vyvvzlqatx6a4oveqkw",
+            "fields":
+            {
+                "title": "[clang-format][NFC] Clean up class HeaderIncludes and Format.cpp",
+                "uri": "https://reviews.llvm.org/D134852",
+                "authorPHID": "PHID-USER-vou2cb5rty2zlopptj5z",
+                "status":
+                {
+                    "value": "published",
+                    "name": "Closed",
+                    "closed": true,
+                    "color.ansi": "cyan"
+                },
+                "repositoryPHID": "PHID-REPO-f4scjekhnkmh7qilxlcy",
+                "diffPHID": "PHID-DIFF-6qic23rkxpwvkp6g4wdg",
+                "summary": "",
+                "testPlan": "",
+                "isDraft": false,
+                "holdAsDraft": false,
+                "dateCreated": 1664433452,
+                "dateModified": 1665032091,
+                "policy":
+                {
+                    "view": "public",
+                    "edit": "users"
+                }
+            },
+            "attachments": {}
+        }
+
+    """
+
+    def __init__(self, data):
+        self.data = data
+        self.title = data["fields"]["title"]
+        self.uri = data["fields"]["uri"]
+
+    def __str__(self):
+        """ String representation """
+        return f"{self.uri} {self.title}"
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#  Stats
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+class DifferentialsCreated(Stats):
+    """ Differentials authored """
+
+    def fetch(self):
+        log.info("Searching for differentials created by %s", self.user)
+        data_dict = {}
+        for idx, login in enumerate(self.parent.phabricator.login_phids):
+            data_dict[f'constraints[authorPHIDs][{idx}]'] = login
+        data_dict['constraints[createdStart]'] = self.options.since.date.strftime(
+            "%s")
+        data_dict['constraints[createdEnd]'] = self.options.until.date.strftime(
+            "%s")
```

Comment:
These can fit on a single line.

Reply:
Sorry, no. I new that I had not formatted this just for fun. The pre-commit is complaining.

![Bildschirmfoto vom 2023-01-20 21-26-35](https://user-images.githubusercontent.com/193408/213799268-0bb0b823-0933-4a43-b0f9-1974fb4cb245.png)

Reply:
I see, sorry that was the default maximum line width. I've updated this in #286 which is now merged. Could you please rebase on the latest `main` content? Thanks.

Reply:
Done

Reply:
Wait, I forgot to adjust things, sigh

Reply:
Thanks. Added a small nitpick in 37274ae.